### PR TITLE
About Page: `text-wrap: balance;`adds unintended new lines in Japaneese

### PR DESCRIPTION
**Pull request by AKSHAT2802**
This is a sample PR to conditionally add text-wrap property in about page CSS based on locale.
https://core.trac.wordpress.org/ticket/60892

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR!

I think the logic is good, but I would like to suggest some tweaks to improve the code quality.

- I think some comment is needed because it is not immediately clear why this code is needed.
- `$curr_locale` variable: Abbreviated variable names don't seem to be used much in WordPress core. How about just `$locale`?
- `$remove_prop_array` variable seems abstract to me. What about `$no_text_wrap_locales`?
- `$additional_class` variable seems abstract to me. What about `$no_text_wrap_class`?
- For non-Japanese locales, there will be a space at the end of the opening tag of the container div: `<div class="wrap about__container ">`.
  Also, since class names are HTML attributes, it would be better to use `esc_attr` rather than `esc_html`.
  In summary, they are as follows:
  `<div class="wrap about__container<?php echo esc_attr( $no_text_wrap_locales ); ?>">`
  Please note that the string to be applied is `' no-text-wrap'`, not `'no-text-wrap'`.
- It is a good idea to add the new CSS to unset text-wrap after `.about__container {h1-h6}` selector to make it clear that it is overriding text-wrap.

**Comment by sabernhardt** [comment]
If a language-specific correction is preferable, that is possible with only CSS:

```
.about__container :is(h1, h2, h3, h4):lang(ja) {
	text-wrap: unset;
}
```

And *if* someone identifies another language that needs the same style (for example, Korean), the languages could be grouped inside another `:is()`.

```
.about__container :is(h1, h2, h3, h4):is(:lang(ja), :lang(ko)) {
	text-wrap: unset;
}
```

The `:is()` selector is [compatible with browsers](https://caniuse.com/css-matches-pseudo) that WordPress admin supports.

**Comment by t-hamano** [comment]
The `:is` selector is probably the simplest and best one, and it certainly works:

image

**Comment by AKSHAT2802** [comment]
Thanks, @t-hamano for the review. 
I have addressed the comments in the new commit.

File: src/wp-admin/css/about.css [hunk 1]
```diff
@@ -411,6 +412,13 @@
 	color: inherit;
 }
 
+.about__container .no-text-wrap h1,
+.about__container .no-text-wrap h2,
+.about__container .no-text-wrap h3,
+.about__container .no-text-wrap h4 {
+	text-wrap: unset;
+}
```
Comment (t-hamano) [inline src/wp-admin/css/about.css]:
```suggestion
.about__container :is(h1, h2, h3, h4):lang(ja) {
	text-wrap: unset;
}
```

@AKSHAT2802 Sorry for the confusion. As [this comment](https://github.com/WordPress/wordpress-develop/pull/7209#issuecomment-2305287152) says, it seems best to solve the issue only with CSS selectors. Could you apply this suggestion and revert the changes to the PHP file?

**Comment by AKSHAT2802** [comment]
Sure @t-hamano, 
I probably missed [comment](https://github.com/WordPress/wordpress-develop/pull/7209#issuecomment-2305287152) the second comment by @sabernhardt. 
Added changes according to the suggestion.

**Review by t-hamano — APPROVED** [review]
Thanks for the update! It works as expected.

I would be happy if we could receive an additional review from @sabernhardt.

| Before | After |
|--------|--------|
| before | after |

File: src/wp-admin/css/about.css [hunk 2]
```diff
@@ -411,6 +411,10 @@
 	color: inherit;
 }
 
+.about__container :is(h1, h2, h3, h4):lang(ja) {
+	text-wrap: unset;
+}
+
```
Comment (ryelle) [inline src/wp-admin/css/about.css:417]: I mentioned in [the core ticket](https://core.trac.wordpress.org/ticket/60892#comment:15) that this could be adjusted in Korean as well, here's that suggestion:

```suggestion

.about__container :is(h1, h2, h3, h4):lang(ko) {
	word-break: keep-all;
}
```

For example, this splits the string at the space between `에 오신`, rather than interrupting `오신`.

| Before | After |
|---|---|
| Screenshot 2024-08-26 at 12 58 51 PM | Screenshot 2024-08-26 at 12 57 01 PM |

**Comment by AKSHAT2802** [comment]
Hii @ryelle @t-hamano waiting for your final review. Addressed all suggested changes. 😃

**Comment by t-hamano** [comment]
@AKSHAT2802 Thanks for the update.

We are discussing in the core ticket what would be a more ideal approach.

Please see https://core.trac.wordpress.org/ticket/60892.

**Comment by hellofromtonya** [comment]
Committed with a few tweaks via https://core.trac.wordpress.org/changeset/58976. Thanks everyone for your contributions 🌟
